augly/video/augmenters/ffmpeg/base_augmenter.py

```diff
@@ -63,3 +66,43 @@ def add_augmenter(
             applied and a dictionary with any output arguments as necessary
         """
         raise NotImplementedError("Implement add_augmenter method")
+
+
+class BaseVidgearFFMPEGAugmenter(ABC):
+    def add_augmenter(
+        self, video_path: str, output_path: Optional[str] = None, **kwargs
+    ) -> None:
+        """
+        Applies the specific augmentation to the video
+
+        @param video_path: the path to the video to be augmented
+
+        @param output_path: the path in which the resulting video will be stored.
+            If not passed in, the original video file will be overwritten
+        """
+        video_path, output_path = validate_input_and_output_paths(
+            video_path, output_path
+        )
+        with tempfile.NamedTemporaryFile(
+            suffix=video_path[video_path.index(".") :]
+        ) as tmpfile:
+            if video_path == output_path:
+                shutil.copyfile(video_path, tmpfile.name)
+                video_path = tmpfile.name
+            writer = WriteGear(output_filename=video_path, logging=True)
```

Comment:
```suggestion
            writer = WriteGear(output_filename=output_path, logging=True)
```

should this be output_path?